File: src/ProjectR.Scripting/content/scripts/spells/lua/OffenseIsTheBestDefense.lua
function GetTargetType()
  return Enemies
end

function GetName()
  return "Offense Is The Best Defense"
end

function GetDescription()
  return "Fast as the wind you attack all enemies and try to PAR them."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .8
end

function GetDelay()
  return .35
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 6 * aAD - 1.5 * dDEF  
  d:ApplyDebuff(PAR, 20)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/OffenseIsTheBestDefense.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class OffenseIsTheBestDefense : SpellScriptBase
    {
        public override string Name { get { return "Offense Is The Best Defense"; } }
        public override string Description { get { return "Fast as the wind you attack all enemies and try to PAR them."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 80; } }
        public override double Delay { get { return .35; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 6 * cAD - 1.5 * tDEF;
            TryToApplyDebuff(DebuffResistance.PAR, 20);
            DealDamage(damage);
        }
    }
}